acra/configuring-maintaining/general-configuration/acra_keymaker.md

```diff
@@ -0,0 +1,178 @@
+---
+title: AcraKeymaker
+bookCollapseSection: true
+---
+
+# AcraKeymaker
+
+## Command line flags
+
+### General flags
+
+* `--client_id=<id>`
+  
+  Use provided client ID as identifier to generate keys or keypairs.
+
+* `--tls_cert=<path>`
+  
+  Path to TLS certificate to use as client_id identifier. Works only when passed empty ClientID: `--client_id=""`.
+
+* `--tls_identifier_extractor_type={distinguished_name|serial_number}`
+  
+  Decide which field of TLS certificate to use as ClientID.
+
+  * `distinguished_name` — (default) certificate Distinguished Name (DN).
+  * `serial_number` — certificate serial number.
+
+* `--keystore=<v1|v2>`
+  
+  Set keystore format. If `--keys_output_dir` points to existing keystore destination then `acra-keymaker` automatically detects version of keystore.
+  If destination storage is empty and hasn't any key then this argument is required. Read more about [keystore versions]({{< ref "/acra/keys/versions/" >}})
+  
+### Storage destination
+
+#### Filesystem
+
+* `--keys_output_dir=<folder>`
+  
+  Folder where will be saved keys (you can change destination for asymmetric public keys with keystore V1 (V2 store at same destination)  with parameter `--keys_public_output_dir`).  
+  Default is `.acrakeys`.
+
+* `--keys_public_output_dir=<folder>`
+  
+  Folder where will be saved public asymmetric keys. Works only for `--keystore=v1` and do nothing with `--keystore=v2`.
+  Default is `.acrakeys`.
+  
+#### Redis
+
+* `--redis_db_keys=`
+
+  Number of Redis database for keys.
+  Default is `0`.
+  <!-- `acra-server -help` says default is `-1` but in `cmd/redis.go` I see `redisDefaultDB = 0` -->
+  <!-- this var is also used as default value for the flag, where's the truth? -->
+
+* `--redis_host_port=<host:port>`
+
+  `<host:port>` used to connect to Redis.
+  Default is empty (don't connect).
+
+* `--redis_password=<password>`
+
+  Password to Redis database.
+
+
+### Configuration files
+
+* `--config_file=<filename>`
+
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-translator.yaml`.
+
+* `--generate_markdown_args_table`
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-translator.md`.
+  Works in pair with `--dump_config`.
+
+### Vault
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `http://x.x.x.x:yyyy`) for loading `ACRA_MASTER_KEY` from HashiCorp Vault.
+  Default is empty (`ACRA_MASTER_KEY` environment variable is expected).
```

Comment:
If you insist on HashiCorp Vault™ branding everywhere, the header should reflect that.

Also, I think it's enough to mention just once how Vault is used by Acra.

Also, using HTTP is bad practice. I don't think Vault recommends that (but IIRC, it *is* possible to disable it).

```suggestion
### HashiCorp Vault

`acra-keymaker` can read `ACRA_MASTER_KEY` from HashiCorp Vault instead of environment variable.

* `--vault_connection_api_string=<url>`

  Connection string (like `https://example.com:8200`) for connecting to HashiCorp Vault.
  If not specified, `ACRA_MASTER_KEY` environment variable will be used.
```